pkg/kp/pcstore/consul_store.go

```diff
@@ -75,9 +82,25 @@ func (s *consulStore) Create(
 	if !success {
 		return fields.PodCluster{}, util.Errorf("Could not set pod cluster at path '%s'", key)
 	}
+
+	// Should we delete the PC if the labels fail?
+	err = s.setLabelsForPC(pc)
+	if err != nil {
+		return fields.PodCluster{}, s.Delete(pc.ID)
```

Comment:
yeah, although now i don't know what to do if delete fails. I guess i'll add a logger and log it. we should probably throw exception reporting integration on these stores as well